### PR TITLE
LPS-143281

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Related Issue: https://issues.liferay.com/browse/LPS-143281

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java [hunk 1]
```diff
@@ -56,50 +55,45 @@
 	public String render(DDMFormFieldValue ddmFormFieldValue, Locale locale) {
 		Value value = ddmFormFieldValue.getValue();
 
-		return render(value.getString(locale), locale);
+		return render(value.getString(value.getDefaultLocale()), locale);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java:58]: @rodrigopaulino why did you change to `value.getDefaultLocale()` when getting the string, but in [line 61](https://github.com/liferay-forms/liferay-portal/pull/1280/files#diff-720a0806d0d1d86d9161878fa05f4753042b94e3bdefc2c89f119244edb075d8R61) you renamed the second parameter to `defaultLocale`? 
`protected String render (String valueString, Locale defaultLocale)`

Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java:58]: Oh! This was just a test I was doing that I forgot to revert. Thanks for asking.

**Comment by rodrigopaulino** [comment]
@carolmariaabb the baseline is now suggesting a major change, but I don't believe it is related to the changes I've made. I guess something was merged that wasn't released thus causing this. So I just pushed the major bump, because it's the only way I can do so tests won't fail. We'll see when it hits Brian's repo.

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java (hunk 1 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java]: I fixed the `value.getDefaultLocal()` argument, and I changed the name of the parameter to `defaultLocale`, because it will only be used if there's no ThemeDisplay's Locale available.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java [hunk 2]
```diff
@@ -56,50 +55,45 @@
...
-				Format format = FastDateFormatFactoryUtil.getSimpleDateFormat(
-					pattern);
-
-				return format.format(
-					DateUtil.parseDate("yyyy-MM-dd", valueString, locale));
-			}
-			catch (ParseException parseException) {
-				_log.error("Unable to parse date", parseException);
-			}
+	private String _render(Locale defaultLocale, String valueString) {
```
Comment (rodrigopaulino) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/date/DateDDMFormFieldValueRenderer.java:61]: @carolmariaabb, I believe you approved the pull before this change. Could you please give your notes on this? Thanks!

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
Rebase with Brian's repo.

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please fix the source formatter issues and resend? Thanks

**Comment by rodrigopaulino** [comment]
ci:reopen

**Comment by rodrigopaulino** [comment]
Failed because Brian's repo has a commit with an invalid message. I'll way for the master sync to rebase this again.

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:reevaluate:1307401_1411
